### PR TITLE
sweetalert upgrade to sweetalert2

### PR DESCRIPTION
**Pull request by Haumer**
please read comments before merging

live demo:
http://g.recordit.co/3xEgF1EC8q.gif

Note I have not adapted it to any other buttons, so check on the other places where you use the alert!

### CODE REVIEW

File: app/javascript/plugins/init_sweetalert.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import swal from 'sweetalert';
+// import swal from 'sweetalert';
+import Swal from 'sweetalert2'
```
Comment (Haumer) [inline app/javascript/plugins/init_sweetalert.js:2]: note the capital S for Swal (sweetalert2)

File: app/javascript/plugins/init_sweetalert.js [hunk 2]
```diff
@@ -20,7 +21,40 @@ const initSweetalert = (selector, options = {}, callback = () => {} ) => {
       const id = swalButton.dataset.id
 
       swalButton.addEventListener('click', () => {
-        swal(options).then(callback(id));
+        // swal(options).then(callback(id));
+        Swal.fire({
```
Comment (Haumer) [inline app/javascript/plugins/init_sweetalert.js:25]: sweetalert2 offers the method .fire()

File: app/javascript/plugins/init_sweetalert.js [hunk 3]
```diff
@@ -20,7 +21,40 @@ const initSweetalert = (selector, options = {}, callback = () => {} ) => {
...
+          confirmButtonText: 'Yes, delete it!',
+          cancelButtonText: 'No, keep it'
+        }).then((result) => {
+          if (result.value) {
+            console.log(result)
+            console.log(result.value)
+            if (result.value) {
+              const link = document.querySelector(`#delete-link-${id}`);
+              link.click();
+              value = false
```
Comment (Haumer) [inline app/javascript/plugins/init_sweetalert.js:39]: delete the item (this is legacy code from your sweetalert)

File: app/javascript/packs/application.js [hunk 4]
```diff
@@ -37,8 +37,8 @@ import { initMapboxTwo } from '../plugins/init_mapbox_two';
 import { initAutocomplete } from '../plugins/init_autocomplete';
 import { initFullCalendar } from '../plugins/init_fullcalendar';
 import { initSweetalert } from '../plugins/init_sweetalert';
-
-
+import Swal from 'sweetalert2/dist/sweetalert2.js'
+import 'sweetalert2/src/sweetalert2.scss'
```
Comment (Haumer) [inline app/javascript/packs/application.js:41]: importing sweetalert2 + style

File: app/javascript/plugins/init_sweetalert.js [hunk 5]
```diff
@@ -20,7 +21,40 @@ const initSweetalert = (selector, options = {}, callback = () => {} ) => {
...
+          } else if (result.dismiss === Swal.DismissReason.cancel) {
+            Swal.fire(
+              'Cancelled',
+              'Your imaginary file is safe :)',
+              'error'
+            )
+          }
+        }).then(
+        callback(id)
+        )
```
Comment (farisalamin) [inline app/javascript/plugins/init_sweetalert.js:57]: What is this callback doing now? It seems like the callback isn't necessary anymore now that the code is all being done above.

Comment (Haumer) [reply app/javascript/plugins/init_sweetalert.js:57]: yeah, you can remove

**Pull request merged**